### DISCUSSION TITLE
Show mongodb slow connection warning after 2 seconds

### DISCUSSION THREAD

**Discussion by thgh** [payloadcms/payload#1372] [Feature Requests & Ideas]
Sometimes it happens mongodb is not running locally and it takes 10 seconds for the `MongooseError: Operation `users.countDocuments()` buffering timed out after 10000ms` error to appear. The DX can be improved by showing a warning when the database seems to be down. The timeout should stay long, as not to break apps where the database connection takes a long time. It should just guide the user to start their database when it seems down. (not sure what a normal connection time is)

**Comment by Rikhart** [comment]
I have the same experience, this happen when the database is not local. A find operation delay is about 7-15 seconds to be resolved. Exist any way to pass more parameters to the connections like autoReconnect=true to the created mongo connection?

**Reply by DanRibbens** [reply]
You can! In your payload.init you can add `mongoOptions`. It's mentioned on this page of the docs: https://payloadcms.com/docs/getting-started/installation

You can also add your connection options as query parameters to the connection string in the ’mongoURL`.

Does that help?
